src/Knp/Menu/Provider/LazyProvider.php

```diff
@@ -0,0 +1,44 @@
+<?php
+
+namespace Knp\Menu\Provider;
+
+/**
+ * A menu provider building menus lazily thanks to builder callables.
+ *
+ * Builders can either be callables or a factory for an object callable
+ * represented as array(Closure, method), where the Closure gets called
+ * to instantiate the object.
+ */
+class LazyProvider implements MenuProviderInterface
+{
+    private $builders;
+
+    public function __construct(array $builders)
```

Comment:
but this is wrong. See the phpdoc of the class. Not all elements of the array are callables.